napari/utils/tree/selection.py

```diff
@@ -0,0 +1,21 @@
+from typing import Iterable, Optional, TypeVar
```

Comment:
Is there anything really Tree Specific here? Should we put it in our events folder so the EventedList can use it one day too :-)